packages/salesforcedx-vscode-core/src/commands/auth/forceAuthWebLogin.ts

```diff
@@ -22,39 +24,136 @@ import {
     );
-    if (isSFDXContainerMode()) {
-      command
-        .withArg('force:auth:device:login')
-        .withLogName('force_auth_device_login');
-    } else {
-      command
-        .withArg('force:auth:web:login')
-        .withLogName('force_auth_web_login');
+
+    command
+      .withArg('force:auth:device:login')
+      .withLogName('force_auth_device_login')
+      .withFlag('--setalias', data.alias)
+      .withFlag('--instanceurl', data.loginUrl)
+      .withArg('--setdefaultusername')
+      .withJson();
+
+    return command.build();
+  }
+
+  public execute(response: ContinueResponse<AuthParams>): void {
+    const startTime = process.hrtime();
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const cancellationToken = cancellationTokenSource.token;
+    const execution = new CliCommandExecutor(this.build(response.data), {
+      cwd: getRootWorkspacePath(),
+      env: { SFDX_JSON_TO_STDOUT: 'true' }
+    }).execute(cancellationToken);
+    let deviceCodeReceived = false;
+
+    channelService.streamCommandStartStop(execution);
+
+    let stdOut = '';
+    execution.stdoutSubject.subscribe(cliResponse => {
+      stdOut += cliResponse.toString();
+
+      if (!deviceCodeReceived) {
+        const authUrl = this.parseAuthUrlFromStdOut(stdOut);
+
+        if (authUrl) {
+          deviceCodeReceived = true;
+          // open the default browser
+          vscode.env.openExternal(vscode.Uri.parse(authUrl, true));
+        }
+      }
+    });
+
+    execution.processExitSubject.subscribe(() => {
+      this.logMetric(execution.command.logName, startTime);
+    });
+
+    notificationService.reportCommandExecutionStatus(
+      execution,
+      cancellationToken
+    );
+
+    ProgressNotification.show(execution, cancellationTokenSource);
+    taskViewService.addCommandExecution(execution, cancellationTokenSource);
+  }
+
+  private parseAuthUrlFromStdOut(stdOut: string): string | undefined {
+    let authUrl;
+    try {
+      const response = JSON.parse(stdOut) as DeviceCodeResponse;
+      const verificationUrl = response.verification_uri;
+      const userCode = response.user_code;
+
+      if (verificationUrl && userCode) {
+        authUrl = `${verificationUrl}?user_code=${userCode}`;
+        this.logToOutputChannel(userCode, verificationUrl);
+      }
+    } catch (error) {
+      channelService.appendLine(
+        nls.localize('force_auth_web_login_device_code_parse_error')
+      );
+      telemetryService.sendException(
+        'force_auth_web_container',
+        `There was an error when parsing the cli response ${error}`
+      );
     }
+
+    return authUrl;
+  }
+
+  private logToOutputChannel(code: string, url: string) {
+    channelService.appendLine(`${EOL}`);
+    channelService.appendLine(nls.localize('action_required'));
+    channelService.appendLine(
+      nls.localize('force_auth_device_login_enter_code', code, url)
+    );
+    channelService.appendLine(`${EOL}`);
+  }
+}
```

Comment:
I added this function to log the same user instructions to the output channel as the CLI. This way if we need to provide the user another way to get the information they need to complete authentication we can direct them to this. @AnanyaJha @gbockus-sf 
Note that the output channel will _not_ open on its own, they will have to get to it manually, as intended.
<img width="1793" alt="Screen Shot 2022-05-04 at 3 35 26 PM" src="https://user-images.githubusercontent.com/25188632/166830743-e1272e9a-8cc7-479b-a460-a3f8085e8a5e.png">